src/middlewared/middlewared/plugins/smb.py

```diff
@@ -14,6 +14,7 @@
 import re
```

Comment:
Not that you have altered the `def do_create` method in this plugin, I'm hitting a `KeyError` using the API directly, because I'm trying to create a SMB share without the `name` attribute. You call the `def clean()` method expecting that the `name` key being preset. Not sure how to handle that scenario? Do you require the name at the schema level? Or do you just `name = data.get('name', '')` in the `clean()` methods.